### DISCUSSION TITLE
Integration testing in an Nx workspace

### DISCUSSION THREAD

**Discussion by benediktms** [prisma/prisma#10792] [Q&A]
Hi there! 

I'm trying to set up an [Nx](https://nx.dev/) workspace that uses a Nestjs backend to serve a GraphQL API. I'm using Postgres as my DB. I've managed to set up prisma for development (reading + writing to the DB works), however I would like to start writing some integration tests and I encountered some issues. I've followed the steps in the [docs](https://www.prisma.io/docs/guides/testing/integration-testing) about integration testing, but there are a few issues when applying this to my scenario and I assume Nx workspaces in general:

Nx promotes the practice of splitting code into reusable libraries. Different libraries can house logic that needs to read and write to the database. They also come with their own commands to run tests, e.g. `yarn nx run feature:test`. Nx also does some caching to determine which part of the code base changes and will only run things like tests on the features that were affected, e.g. `nx run affected --target=test`. This is what I use (or would like to) to run tests for CI. This is where things start to get tricky. When the Postgres instance used for integration tests runs in a docker container migrations have to be applied before the tests run, and after wards the container has to be destroyed/restarted. 

My issue is in getting Nx to play nice with prisma's migrate commands: Prefixing the command to something like `docker-compose up -d && npx prisma migrate deploy && nx run affected --target=test && docker-compose down` will enter a never ending loops of starting the docker containers. When trying this manually, i.e. running `docker-compose up -d && npx prisma migrate deploy && nx run feature:test && docker-compose down` things work a *bit* better. I can manage to initially write to the database, however the container does not get destroyed after the test. I've tried `docker remove ...` to manually destroy the container, but I'm not getting much success.

Alternatively, I was toying with the idea of using sqlite's `:memory:` feature to run all tests against an sqlite DB in memory, but ideally I'd like to test against Postgres.

I'd love to hear how the community handles integration testing with prisma, ideally within an Nx workspace.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
